### PR TITLE
Add rule to detect duplicate block names

### PR DESCRIPTION
**Pull request by Christian-Rades**

### CODE REVIEW

File: crates/ludtwig/src/check/rule.rs [hunk 1]
```diff
@@ -150,13 +159,27 @@ pub struct CheckNote {
     pub message: String,
 }
 
+#[derive(Debug)]
+pub enum SecondaryNote {
```
Comment (Christian-Rades) [inline crates/ludtwig/src/check/rule.rs:163]: Not quite sure if this a the best solution, but it gets the job done :thinking:

File: crates/ludtwig/src/check/rules/duplicate_block_names.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+mod tests {
+    use expect_test::expect;
+
+    use crate::check::rules::test::test_rule;
+
+    #[test]
+    fn rule_reports() {
+        test_rule(
+            "duplicate-block-names",
+            r#"{% block foo %}{% endblock %} {%block foo%} {% endblock %}"#,
```
Comment (Christian-Rades) [inline crates/ludtwig/src/check/rules/duplicate_block_names.rs:73]: Happy to get some Ideas for edge cases

File: crates/ludtwig/src/check/rule.rs (hunk 1 above)
Comment (MalteJanz) [reply crates/ludtwig/src/check/rule.rs:163]: I would rather prefer another member in the check result struct like so
```rust
#[derive(Debug)]
pub struct CheckResult {
    // file_id
    pub(super) rule_name: &'static str,
    pub(super) severity: Severity,
    pub(super) message: String,
    pub(super) primary: Option<CheckNote>,
    pub(super) secondary: Vec<CheckNote>,
    pub(super) suggestions: Vec<CheckSuggestion>,
}
```
Also the `CheckHint` struct is basically the same as the already existing `CheckNote` struct 😉 

But no worries, I can also adjust this if I find the time somewhere in the future 😅

File: crates/ludtwig/src/check/rules/duplicate_block_names.rs (hunk 2 above)
Comment (MalteJanz) [reply crates/ludtwig/src/check/rules/duplicate_block_names.rs:73]: This rule implementation is much cleaner than I expected, great job 👍 .
I don't know any specific edge cases to test, except maybe one:

You are iterating the tree by yourself and therefore must uphold the following invariants:
- Don't look into Error Syntax Nodes ✅ 
- Check for `ludtwig-ignore` statements before your inspected syntax node ❌ 
The latter is kinda stupid because why would you want to ignore this rule? But for completeness it should be possible to ignore it 😂 . For inspiration this is also done in the indentation rule (see function `check_for_rule_ignore_enter`)

**Review by MalteJanz — CHANGES_REQUESTED** [review]

**Comment by tobiasberge** [comment]
@MalteJanz @Christian-Rades Not sure if we need to cover this, for me a duplicate block declaration already causes a hard twig exception. 🤔 

Test in base.html.twig of sw platform:
```diff
{% block base_body %}
    <body class="{% block base_body_classes %}is-ctl-{% if controllerName is not empty %}{{ controllerName|lower }}{% endif %} is-act-{% if controllerAction is not empty %}{{ controllerAction|lower }}{% endif %}{% endblock %}">

+    {% block base_body_inner %}{% endblock %}

    {% block base_body_inner %}
        {% block base_noscript %}
```
It does kaboom for me.

**Comment by Christian-Rades** [comment]
@tobiasberge The idea is to find these errors without running the code. 
I suspect you still need to attempt to render the template to get the exception.

**Comment by MalteJanz** [comment]
@tobiasberge Sure this specific rule does not provide many benefits (because the symfony twig linter also checks for duplicated twig block names). But I would still include it for the sake of completeness and maybe nicer error messages 😄 

Also this could be really beneficial for our Administration Twig.js templates (if ludtwig gets mature enough and maybe some day finds it's way back to the Twig.js templates...). Because there is currently no Twig validation at all 🙈

**Review by MalteJanz — APPROVED** [review]
